vue-components/src/components/Button.vue

```diff
@@ -64,83 +75,169 @@ $base: '.wikit-Button';
 
 	&:not(:disabled) {
 		&#{$base}--neutral {
-			color: $wikit-Button-neutral-color;
-			background-color: $wikit-Button-neutral-background-color;
-			border-color: $wikit-Button-neutral-border-color;
+			color: $wikit-Button-framed-neutral-color;
+			background-color: $wikit-Button-framed-neutral-background-color;
+			border-color: $wikit-Button-framed-neutral-border-color;
+
+			&:hover {
+				color: $wikit-Button-framed-neutral-hover-color;
+				background-color: $wikit-Button-framed-neutral-hover-background-color;
+				border-color: $wikit-Button-framed-neutral-hover-border-color;
+			}
+
+			&:active {
+				color: $wikit-Button-framed-neutral-active-color;
+				background-color: $wikit-Button-framed-neutral-active-background-color;
+				border-color: $wikit-Button-framed-neutral-active-border-color;
+			}
+
+			// A clicked button is both :active and :focused. Using :not(:active) to avoid mixing the two.
+			&:focus:not(:active) {
+				color: $wikit-Button-framed-neutral-focus-color;
+				background-color: $wikit-Button-framed-neutral-focus-background-color;
+				border-color: $wikit-Button-framed-neutral-focus-border-color;
+				box-shadow: $wikit-Button-framed-neutral-focus-box-shadow;
+			}
+		}
+
+		&#{$base}--quiet--neutral {
```

Comment:
I think the `--quiet` modifier should become its own class (`.wikit-Button--quiet`) instead of being added on top of the Button's type modifier. That makes the structure clearer, especially once we add the normal/framed button variant. It allows us to share styles between button variants (quiet, primary, normal).

I imagine the styles could be arranged in a way similar to this:
```scss
#{$base} {
	// styles shared between all buttons
	&--quiet {
		// styles shared between quiet buttons if any
		&#{$base}--normal {
			// ...
		}
		&#{$base}--progressive {
			// ...
		}
		&#{$base}--destructive {
			// ...
		}
	}

	&--primary {
		// styles shared between primary buttons if any

		&#{$base}--progressive {
			// ...
		}
		&#{$base}--destructive {
			// ...
		}
	}

	&--normal {
		// styles shared between normal buttons if any
		&#{$base}--neutral {
			// ...
		}
	}
}
```

Reply:
Hey, Jakob! We had a catch up with the hike regarding exactly this, and we agreed to this structure, but tentatively decided (please let us know what you think) that it would make more sense to restructure the button classes and modifiers in a different PR. I just opened a task in the DS board: https://phabricator.wikimedia.org/T270397

Reply:
I don't see how what's in this PR right now is an intermediate step leading towards what's outlined in the ticket. The styles could already reflect this structure now without requiring changes to the component's props.

Reply:
On the other hand I can see how it becomes ugly with `primaryProgressive` and `primaryDestructive` currently encoding both the type and the variant in one. In that case it actually seems more sensible to restructure the props and styles *before* adding the quiet variant.